src/Libraries/CoreNodeModels/Input/BaseTypes.cs

```diff
@@ -103,12 +103,29 @@ public DoubleInput()
 
             ShouldDisplayPreviewCore = false;
             ConvertToken = Convert;
-            Value = "0";
+            NumericalValue = "0";
 
             //ws.DynamoModel.PreferenceSettings.PropertyChanged += Preferences_PropertyChanged;
         }
 
-        public virtual double Convert(double value)
+        //if the string can be parsed to a common numeric type return true
+        private bool validateInput(string value)
+        {
+            bool canConvert = false;
+
+            int intVal;
+            var canConvertInt = int.TryParse(value, out intVal);
+            double doubleVal;
+            var canConvertDouble = double.TryParse(value, out doubleVal);
+            long longVal;
+            var canConvertLong = long.TryParse(value, out longVal);
```

Comment:
You should use CultureInfo.InvariantCulture here.  We had issues around this earlier.